static/app/views/replays/deadRageClick/constructSelector.tsx

```diff
@@ -5,32 +5,48 @@ function trimAttribute(elementAttribute, fullAlltribute) {
 }
 
 export default function constructSelector(element: ReplayClickElement) {
-  const fullAlt = '[alt="' + element.alt + '"]';
-  const alt = trimAttribute(element.alt, fullAlt);
+  const tag = element.tag;
 
-  const fullAriaLabel = '[aria="' + element.aria_label + '"]';
-  const ariaLabel = trimAttribute(element.aria_label, fullAriaLabel);
+  const id = trimAttribute(element.id, '#' + element.id);
 
   const trimClass = element.class.filter(e => e !== '');
   const classWithPeriod = trimClass.join('.');
   const classNoPeriod = classWithPeriod.replace('.', '');
   const classes = trimAttribute(classNoPeriod, '.' + classWithPeriod);
 
-  const id = trimAttribute(element.id, '#' + element.id);
+  const fullAlt = '[alt="' + element.alt + '"]';
+  const alt = trimAttribute(element.alt, fullAlt);
+
+  const fullAriaLabel = '[aria="' + element.aria_label + '"]';
+  const ariaLabel = trimAttribute(element.aria_label, fullAriaLabel);
 
   const fullRole = '[role="' + element.role + '"]';
   const role = trimAttribute(element.role, fullRole);
 
-  const tag = element.tag;
-
   const fullTestId = '[data-test-id="' + element.testid + '"]';
   const testId = trimAttribute(element.testid, fullTestId);
 
   const fullTitle = '[title="' + element.title + '"]';
   const title = trimAttribute(element.title, fullTitle);
 
+  const fullComponentName = '[data-sentry-component="' + element.component_name + '"]';
+  const componentName = trimAttribute(element.component_name, fullComponentName);
+
   const fullSelector =
-    tag + id + classes + fullRole + fullAriaLabel + fullTestId + fullAlt + fullTitle;
-  const selector = tag + id + classes + role + ariaLabel + testId + alt + title;
-  return {fullSelector, selector};
+    tag +
+    id +
+    classes +
+    fullRole +
+    fullAriaLabel +
+    fullTestId +
+    fullAlt +
+    fullTitle +
+    fullComponentName;
+  const selector =
+    tag + id + classes + role + ariaLabel + testId + alt + title + componentName;
+  const displaySelector = componentName
+    ? element.component_name + id + role + ariaLabel + testId + alt + title
+    : selector;
+
+  return {fullSelector, selector, displaySelector};
```

Comment:
could we add a comment somewhere in this file to explain the differences between `fullSelector`, `selector`, and `displaySelector`? just for sanity, because the list is getting big now